pkg/clusteragent/admission/metrics/metrics.go

```diff
@@ -43,7 +43,7 @@ var (
 		[]string{"mutation_type", "reason", "language", "auto_detected"}, "Number of mutation failures by mutation type (agent config, standard tags, lib injection).",
 		telemetry.Options{NoDoubleUnderscoreSep: true})
 	WebhooksReceived = telemetry.NewCounterWithOpts("admission_webhooks", "webhooks_received",
-		[]string{}, "Number of mutation webhook requests received.",
+		[]string{"mutation_webhook"}, "Number of mutation webhook requests received.",
```

Comment:
In `MutationAttempts` and `MutationErros`, the same webhook sets different values for the `mutation_type` tag. 

For example, in CWS instrumentation webhook, we set the mutation_type with two different values:
- [here](https://github.com/DataDog/datadog-agent/blob/85871eadd769876875e715f71634ff638caae649/pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go#L280)
- and [here](https://github.com/DataDog/datadog-agent/blob/85871eadd769876875e715f71634ff638caae649/pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go#L365)

I think only CWS instrumentation does this.
Other webhooks use the same value for this tag always.

One other point is that webhooks don't use webhook name as a value for this tag. They instead use some defined constants that are defined [here](https://github.com/DataDog/datadog-agent/blob/85871eadd769876875e715f71634ff638caae649/pkg/clusteragent/admission/metrics/metrics.go#L21-L25). 

Not sure if we can remove these constants and use the webhook name instead. If we do this, I am concerned that some customer is filtering by the old tag value, and then they stop getting metrics with the old tag value.
Another thing that might be possible is to add `mutation_webhook` tag also to `MutationAttempts` and `MutationErrors`, in this way we don't touch the existing tag, although it might be a bit confusing I think to have `mutation_type` and `mutation_webhook`.

WDYT?

Reply:
I think that it's a bit confusing that we're not using the webhook name in the label, but we cannot change it now because as you said it would be a breaking change.

I think that what we should do is use the same label always, `mutation_type`. I think having 2 labels for the same thing is confusing. For the values, we can keep using the ones defined in the constants, and for new webhooks, use the name instead. We can define a map with the webhook_name => label_value relationships if needed.

CWS sets 2 different values because it defines 2 webhooks.

Reply:
I agree that this can work, but I feel like it is making the code twisted in order to achieve something simple.

I thought about two alternatives:

- Changing the webhook names to use the tag values already defined. In this way we can use the webhook names and not cause a breaking change. And this change should not be an issue since I don't think the webhook name is used anywhere (yet).

- Alternatively, we can add a GetMutationTypeTag() method to the webhook interface, and use it as a tag value, and we can re-use the existing tag values to preserve backward compatibility.

WDYT?

Reply:
The name is only used in the mutating webhook created in Kubernetes so it should be OK to change it :+1:
I think that's a better solution that adding a new function to the interface just for this.